### PR TITLE
Splitting loss new demo, SSDU, Artifact2Artifact, Phase2Phase

### PR DESCRIPTION
**Pull request by Andrewwango**
- Change splitting loss demo to use Tomography
- Add Gaussian splitting mask for 2D and 2D+t splitting (for Yaman et al. (SSDU) and Acar et al.)
- Add Artifact2Artifact (from [RARE](https://arxiv.org/abs/1912.05854)) and [Phase2Phase](https://journals.lww.com/investigativeradiology/pages/articleviewer.aspx?year=2021&issue=12000&article=00004&type=Fulltext)
- Fix metrics bug in Trainer
- Add time-agnostic network wrapper

### Checks to be done before submitting your PR
- [x] `python3 -m pytest tests/` runs successfully.
- [x] `black .` runs successfully.
- [x] `make html` runs successfully (in the `docs/` directory).
- [x] Updated docstrings related to the changes (as applicable).
- [x] Added an entry to the [CHANGELOG.rst](../CHANGELOG.rst).

### CODE REVIEW

File: deepinv/tests/test_models.py [hunk 1]
```diff
@@ -560,3 +560,11 @@ def test_optional_dependencies(denoiser, dep):
 #     mse_out = (x_net - x).pow(2).mean()
 #
 #     assert mse_out < mse_in
+
+
+def test_time_agnostic_net():
+    backbone_net = dinv.models.UNet(scales=2)
+    net = dinv.models.TimeAgnosticNet(backbone_net)
+    y = torch.rand(1, 1, 4, 8, 8)  # B,C,T,H,W
```
Comment (tachella) [inline deepinv/tests/test_models.py:568]:
```suggestion
    y = torch.rand(1, 1, 2, 4, 4)  # B,C,T,H,W
```

File: deepinv/training/testing.py [hunk 2]
```diff
@@ -143,18 +145,27 @@ def test(
...
-            if hasattr(physics_cur, "A_adjoint"):
-                if isinstance(physics_cur, torch.nn.DataParallel):
-                    x_init = physics_cur.module.A_adjoint(y)
-                else:
-                    x_init = physics_cur.A_adjoint(y)
-            elif hasattr(physics_cur, "A_dagger"):
-                if isinstance(physics_cur, torch.nn.DataParallel):
-                    x_init = physics_cur.module.A_dagger(y)
+            phys = (
+                physics_cur.module
```
Comment (tachella) [inline deepinv/training/testing.py:149]: I'm going to integrate all this in future PR

**Review by tachella — CHANGES_REQUESTED** [review]
Many thanks for the PR!

I think most of my comments were about esthetic changes, except for

- Cloning the mask in the splittingloss to avoid torch backward issues
- Adapting splitting loss to different measurement sizes at eval time
- We should also test the new splitting-based losses (Phase2Phase and Artifact2Artifact).

File: deepinv/training/testing.py (hunk 2 above)
Comment (Andrewwango) [reply deepinv/training/testing.py:149]: Should I just leave this as is and then this will be overwritten in #280

File: deepinv/loss/measplit.py [hunk 3]
```diff
@@ -209,8 +242,6 @@ def forward(
                 device=y.device,
             )
```
[the remark this answers is not in the document]
Comment (Andrewwango) [reply deepinv/loss/measplit.py:244]: I've added `self.mask_generator.tensor_size != y.size()[1:]` as a condition:
`if self.mask_generator is None or self.mask_generator.tensor_size != y.size()[1:]`
but not added the rest of the suggested code as it's the same as before. Can you check this is ok @bemc22 ?

**Review by tachella — APPROVED** [review]
